zzre/game/systems/ui/ScrBookMenu.cs

```diff
@@ -42,6 +42,30 @@ protected override void HandleOpen(in messages.ui.OpenBookMenu message)
             .Build();
 
         CreateTopButtons(preload, entity, inventory, IDOpenFairybook);
+        CreateFairyButtons(preload, entity, inventory);
+    }
+
+    private void CreateFairyButtons(UIPreloader preload, in DefaultEcs.Entity entity, Inventory inventory)
+    {
+        var fairies = db.Fairies.OrderBy(fairyRow => fairyRow.CardId.EntityId).ToArray();
+        for (int i = 0; i < fairies.Length; i++)
+        {
+            var fairyRow = fairies[i];
+            foreach (var ownedFairy in inventory.Fairies)
+            {
+                if (ownedFairy.cardId == fairyRow.CardId)
+                {
```

Comment:
Instead of looping through yourself you can use `inventory.Contains(fairies[i].CardId)`

Reply:
thank you, fixed